crates/chia-consensus/src/multiprocess_validation.rs

```diff
@@ -0,0 +1,243 @@
+use crate::consensus_constants::ConsensusConstants;
+use std::sync::{Arc, Mutex};
+use std::thread;
+use crate::gen::condition_tools::pkm_pairs;
+use crate::gen::flags::{
+    AGG_SIG_ARGS, ALLOW_BACKREFS, ENABLE_MESSAGE_CONDITIONS, ENABLE_SOFTFORK_CONDITION,
+    NO_RELATIVE_CONDITIONS_ON_EPHEMERAL,
+};
+use crate::gen::owned_conditions::OwnedSpendBundleConditions;
+use crate::gen::solution_generator::solution_generator;
+use crate::gen::validation_error::ErrorCode;
+use crate::generator_types::BlockGenerator;
+use crate::npc_result::get_name_puzzle_conditions;
+use chia_bls::aggregate_verify;
+use chia_bls::BlsCache;
+use chia_bls::PublicKey;
+use chia_protocol::Coin;
+use chia_protocol::Program;
+use chia_protocol::SpendBundle;
+use chia_traits::Streamable;
+use clvmr::{ENABLE_BLS_OPS_OUTSIDE_GUARD, ENABLE_FIXED_DIV};
+use std::time::{Duration, Instant};
+
+// currently in multiprocess_validation.py
+// called via blockchain.py from full_node.py when a full node wants to add a block or batch of blocks
+fn pre_validate_blocks_multiprocessing() {}
+
+// currently in multiprocess_validation.py
+// called in threads from pre_validate_blocks_multiprocessing
+fn batch_pre_validate_blocks() {}
```

Comment:
Yeah I have been / will be working on these in a separate PR. I need some of the same prerequisites such as NPCResult so I may wait for this PR to land so I can rebase.

Alternatively this could be split up into multiple PRs